modules/light-clients/07-tendermint/types/update.go

```diff
@@ -283,6 +250,85 @@ func (cs *ClientState) ValidateClientMessage(
 	return nil
 }
 
+// CheckForMisbehaviour returns true with an optional error if a misbehaviour is found for the given Header or Misbehaviour type
+func (cs ClientState) CheckForMisbehaviour(
+	ctx sdk.Context, cdc codec.BinaryCodec, clientStore sdk.KVStore,
+	msg exported.ClientMessage,
+) (bool, error) {
```

Comment:
Returning an optional error here to allow for more fine-grained error handling (see Misbehaviour section). Not sure if this is golang best practice. 

I think it would be good to sync on our function signatures for these new functions @damiannolan @colin-axner @AdityaSripal

Reply:
`VerifyClientMessage` should return the error. We can panic here (as that implies VerifyClientMessage wasn't called first)